src/stavemodifier.ts

```diff
@@ -0,0 +1,147 @@
+// [VexFlow](http://vexflow.com) - Copyright (c) Mohit Muthanna 2010.
+//
+// ## Description
+// A base class for stave modifiers (e.g. clefs, key signatures)
+
+import { Element } from './element';
+import { Glyph } from './glyph';
+import { Stave } from './stave';
+import { Metrics } from './note';
+export interface LayoutMetrics {
+  xMin: number;
+  xMax: number;
+  paddingLeft: number;
+  paddingRight: number;
+}
+
+export interface StaveModifierSpacer {
+  getContext(): boolean;
+  setStave(): void;
+  renderToStave(): void;
+  getMetrics(): Metrics;
+}
+
+export enum Position {
+  CENTER = 0,
+  LEFT = 1,
+  RIGHT = 2,
+  ABOVE = 3,
+  BELOW = 4,
+  BEGIN = 5,
+  END = 6,
+}
+
+export class StaveModifier extends Element {
+  protected width: number = 0;
+  protected x: number = 0;
+
+  protected padding: number;
+  protected position: Position;
+  protected stave?: Stave;
+  protected layoutMetrics?: LayoutMetrics;
+  static get Position(): typeof Position {
+    return Position;
+  }
+
+  constructor() {
+    super();
+    this.setAttribute('type', 'StaveModifier');
+
+    this.padding = 10;
+    this.position = StaveModifier.Position.ABOVE;
+  }
+
+  getPosition(): number {
+    return this.position;
+  }
+
+  setPosition(position: number): this {
+    this.position = position;
+    return this;
+  }
+  getStave(): Stave | undefined {
+    return this.stave;
+  }
+
+  setStave(stave: Stave): this {
+    this.stave = stave;
+    return this;
+  }
+
+  getWidth(): number {
+    return this.width;
+  }
+
+  setWidth(width: number): this {
+    this.width = width;
+    return this;
+  }
+
+  getX(): number {
+    return this.x;
+  }
+
+  setX(x: number): this {
+    this.x = x;
+    return this;
+  }
+
+  getCategory(): string {
+    return '';
+  }
+
+  makeSpacer(padding: number): StaveModifierSpacer {
```

Comment:
Let's remove this  method and the `StaveModifierSpacer` -- it's not used anywhere.

Reply:
done